examples/all-clusters-app/all-clusters-common/src/binding-handler.cpp

```diff
@@ -102,12 +103,17 @@ static void BoundDeviceChangedHandler(const EmberBindingTableEntry & binding, ch
     }
 }
 
-CHIP_ERROR InitBindingHandlers()
+static void InitBindingHandlerInternal(intptr_t arg)
 {
     chip::BindingManager::GetInstance().SetAppServer(&chip::Server::GetInstance());
     chip::BindingManager::GetInstance().RegisterBoundDeviceChangedHandler(BoundDeviceChangedHandler);
 #if defined(ENABLE_CHIP_SHELL)
     RegisterSwitchCommands();
 #endif
+}
+
+CHIP_ERROR InitBindingHandlers()
+{
+    chip::DeviceLayer::PlatformMgr().ScheduleWork(InitBindingHandlerInternal);
```

Comment:
Please document why this init is being done async. Same for the other places where it's being done.

In general, this looks suspicious: what if something tries to use bindings before that scheduled work happens?

Reply:
Comments added. We have to make the initialization async because establishing connection after loading the binding table require the `Server` instance to be fully initialized first. I didn't find a proper place for the initialization. The ZCL init callback is called before server initialization. I cannot place binding related functions in the server instance since some devices don't have binding at all. As a result I post the initialization to the event queue.